src/com/google/cose/Ec2SigningKey.java

```diff
@@ -114,6 +118,74 @@ public ECPublicKey getPublicKey() {
     return (ECPublicKey) this.keyPair.getPublic();
   }
 
+  private static byte[] arrayFromBigNum(BigInteger n, int curveSize) {
+    byte[] rgb = new byte[(curveSize + 7) / 8];
+    byte[] rgb2 = n.toByteArray();
+    if (rgb.length == rgb2.length) {
+      return rgb2;
+    }
+    if (rgb2.length > rgb.length) {
+      System.arraycopy(rgb2, rgb2.length - rgb.length, rgb, 0, rgb.length);
+    } else {
+      System.arraycopy(rgb2, 0, rgb, rgb.length - rgb2.length, rgb2.length);
+    }
+    return rgb;
+  }
+
+  /** Generates a COSE formatted Ec2 signing key given a specific algorithm */
+  public static Ec2SigningKey generateKey(Algorithm algorithm) throws CborException, CoseException {
+    KeyPair keyPair;
+    int curveSize;
+    int header;
+    String curveName;
+
+    switch (algorithm) {
+      case SIGNING_ALGORITHM_ECDSA_SHA_256:
+        curveName = "secp256r1";
+        curveSize = 256;
+        header = Headers.CURVE_EC2_P256;
+        break;
+
+      case SIGNING_ALGORITHM_ECDSA_SHA_384:
+        curveName = "secp384r1";
+        curveSize = 384;
+        header = Headers.CURVE_EC2_P384;
+        break;
+
+      case SIGNING_ALGORITHM_ECDSA_SHA_512:
+        curveName = "secp521r1";
+        curveSize = 521;
```

Comment:
key size for P-521 is actually 512 bits.. so it should be okay to use 512 curve size here since I think that is only being used to derive the key bits in proper format.

Reply:
Also.. why don't we just rename the variable key size instead of curve size.

Reply:
renamed to keySize, done.